packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1389,8 +1387,21 @@ export class SnapController extends BaseController<
     return state ? this.#decryptSnapState(snapId, state) : null;
   }
 
+  /**
+   * Get a deterministic encryption key based on the given snap ID.
+   *
+   * @param snapId - The ID of the snap to get the encryption key for.
+   * @returns The encryption key for the given snap ID.
+   */
   async #getEncryptionKey(snapId: SnapId): Promise<string> {
-    return this.#getAppKey(snapId, AppKeyType.StateEncryption);
+    const mnemonicPhrase = await this.#getMnemonicPhrase();
+
+    return deriveEntropy({
+      mnemonicPhrase,
+      input: snapId,
+      salt: 'state encryption',
```

Comment:
Could this be a constant somewhere? Do we even need a salt?

Reply:
I added it so we can generate different entropy for other purposes in the future.